core/services/sleeper_task.go

```diff
@@ -37,14 +44,25 @@ func NewSleeperTask(worker Worker) SleeperTask {
 
 // Start begins the SleeperTask
 func (s *sleeperTask) Start() error {
+	s.workerWG.Add(1)
 	go s.workerLoop()
 	return nil
 }
 
 // Stop stops the SleeperTask
 func (s *sleeperTask) Stop() error {
 	s.closer <- struct{}{}
-	return nil
+	ch := make(chan bool, 1)
```

Comment:
Wanna replace the workerWG with this now? Can also be `chan struct{}` without a buffer size since you're only closing it.

Also, what does the race detector say on this?

Reply:
I don't want to replace anything. This ensures that the task exits within 5 second or we return an error.

I don't know how to use the race detector.

Reply:
Right, but we're using a wait group to signal a channel to signal that the go routine is finished, why not just use the channel on its own?

Reply:
Yeah, this is the thing I dislike the most about WaitGroups.  They don't play well with `select`, which means they don't really play all that well with Go's concurrency primitives, so they only make sense if you really need to collapse a fanout.  I say we go with pure channels in this particular case.  Does the channels-only example I posted in Slack pass tests?  We can add this 5 second timeout to that easily (although we should probably use a `context.WithTimeout` for that).

Reply:
This is the original solution we had. It worked fine and solved the problem. I think @rupurt changed it to use a WaitGroup to be more idiomatic.

I will reinstate the pure channel solution :+1: